modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/AbstractExecutionTest.java

```diff
@@ -69,6 +81,22 @@
     /** */
     protected int nodesCnt = 3;
 
+    /** */
+    @Parameterized.Parameters(name = "Execution direction = {0}")
+    public static List<Object[]> parameters() {
+        ArrayList<Object[]> params = new ArrayList<>();
+
+        params.add(new Object[]{true});
+        params.add(new Object[]{false});
+        params.add(new Object[]{null});
+
+        return params;
+    }
+
+    /** Execution direction. */
+    @Parameterized.Parameter
+    public static Boolean execDir;
```

Comment:
May be enum usage will be more appropriate?

Reply:
can`t use enum here, need to be iterable as i understand, does i missed something?

Reply:
@zstan please see my suggestion here: https://github.com/gridgain/apache-ignite/pull/259.

Also introducing params in the abstract class breaks ContinuousExecutionTest

Reply:
thanks ! done.